### PR TITLE
Issue 0107: Update for spyro Compatibility with Latest Firedrake Installation

### PR DESCRIPTION
**Pull request by Olender**
**Description:**

This minor update ensures that spyro is compatible with the latest Firedrake installation. The main changes include:

1. **Function to Cofunction Conversion:** Updated several functions to cofunctions where necessary.
2. **Cleanup:** Removed old, unused test and HTML files that were unnecessarily taking up space in the repository.

### CODE REVIEW

**Comment by Olender** [comment]
I have also updated the runner file to the correct location of the newest Firedrake installation

**Comment by SouzaEM** [comment]
The cell type of some tests was changed from "Q" to "T". Is this a limitation of the newer Firedrake version?

**Comment by Olender** [comment]
Not anymore, the newer version of Firedrake now supports both triangles and quadrilaterals. Initially, I switched to using triangles during testing, as Firedrake had some issues with quadrilaterals at the time. Once these problems were resolved, I chose to continue using triangles since they are our usual preference. However, we still have numerous tests based on quadrilaterals.

File: spyro/solvers/time_integration_central_difference.py [hunk 1]
```diff
@@ -277,9 +281,9 @@ def central_difference_MMS(Wave_object, source_id=0):
 
         B = fire.assemble(rhs, tensor=B)
 
-        Wave_object.solver.solve(X, B)
+        Wave_object.solver.solve(u_np1, B)
 
-        u_np1.assign(X)
+        # u_np1.assign(X)
```
Comment (SouzaEM) [inline spyro/solvers/time_integration_central_difference.py:286]: I would remove commented code. They make code harder to read and make it harder to search for word occurrences.

File: spyro/solvers/time_integration_central_difference.py [hunk 2]
```diff
@@ -241,9 +242,12 @@ def central_difference_MMS(Wave_object, source_id=0):
 
     u_nm1 = Wave_object.u_nm1
     u_n = Wave_object.u_n
+    u_np1 = Wave_object.u_np1
     u_nm1.assign(Wave_object.analytical_solution(t - 2 * dt))
     u_n.assign(Wave_object.analytical_solution(t - dt))
-    u_np1 = fire.Function(Wave_object.function_space, name="pressure t +dt")
+    # u_np1 = fire.Function(Wave_object.function_space, name="pressure t +dt")
```
Comment (SouzaEM) [inline spyro/solvers/time_integration_central_difference.py:248]: I would remove commented code. They make code harder to read and make it harder to search for word occurrences.

File: spyro/solvers/acoustic_solver_construction_no_pml.py [hunk 3]
```diff
@@ -46,6 +48,9 @@ def construct_solver_or_matrix_no_pml(Wave_object):
     Wave_object.solver = fire.LinearSolver(
         A, solver_parameters=Wave_object.solver_parameters
     )
+    # lin_var = fire.LinearVariationalProblem(lhs, rhs + B, u_np1)
```
Comment (SouzaEM) [inline spyro/solvers/acoustic_solver_construction_no_pml.py:51]: I would remove commented code. They make code harder to read and make it harder to search for word occurrences.

**Review by SouzaEM — APPROVED** [review]

File: spyro/solvers/time_integration_central_difference.py (hunk 2 above)
Comment (Olender) [reply spyro/solvers/time_integration_central_difference.py]: Removed

File: spyro/solvers/time_integration_central_difference.py (hunk 1 above)
Comment (Olender) [reply spyro/solvers/time_integration_central_difference.py]: Removed

File: spyro/solvers/acoustic_solver_construction_no_pml.py (hunk 3 above)
Comment (Olender) [reply spyro/solvers/acoustic_solver_construction_no_pml.py]: Removed

**Pull request merged**